internal/beater/otlp/grpc.go

```diff
@@ -79,7 +54,17 @@ func RegisterGRPCServices(
 		Semaphore:        semaphore,
 		RemapOTelMetrics: true,
 	})
-	gRPCMonitoredConsumer.set(consumer)
+
+	// FIXME we should add an otel counter metric directly in the
```

Comment:
I think this needs fixing before we merge anything?

Reply:
> this shouldn't be needed now that we propagate the meterprovider instead of reusing a global meter.

I don't follow. The issue here is that:
 - the MeterProvider & Meters will survive across config reloads
 - on each config reload we will register a new callback
 - we never unregister those callbacks, so memory & CPU usage will increase with every reload

> I also think adding metrics to apm-data is not as simple since we need to keep other systems in mind (that's a library).

What's difficult about that? We can add a MeterProvider to apm-data/input/otlp.ConsumerConfig, and if it's not set then we can either use the global or a noop provider.

The alternative to adding the metric to apm-data would be to find a way to unregister the callback when the server is stopped.

Reply:
> We cannot add apm-server specific metrics in apm-data. That's library and it could be used by other applications.
> If we add a metric to apm-data it should be agnostic but at that point we can't use the same name as the old beats monitoring metric.

Right, I see. I had in mind we'd do something similar to go-docappender: apm-data native metrics that we rename/transform to the legacy apm-server equivalents.

> Moreover if we add metrics to otlp consumer in apm-data we should consistently add all the consumer metrics, not just unsupported_dropped.

I agree we should have that eventually, but we can add them incrementally.

Anyway, since if we can unregister the callback it's fine if we do this later if at all.